src/askui/models/ui_tars_ep/parser.py

```diff
@@ -15,55 +17,99 @@ def parse(cls, coord_str: str) -> "BoxCoordinate":
             raise ValueError(f"Invalid coordinate format: {coord_str}")
         return cls(x=int(match.group(1)), y=int(match.group(2)))
 
-class ClickAction(BaseModel):
-    """Click action with start box coordinates."""
+class BaseAction(BaseModel):
+    action_type: str
+
+    def execute(self, agent_os: AgentOs) -> None:
+        raise NotImplementedError(f"Action '{self.action_type}' must implement execute method.")
+
+        
+class ClickAction(BaseAction):
     action_type: Literal["click"] = "click"
     start_box: BoxCoordinate
 
-class DoubleClickAction(BaseModel):
-    """Double left click action with start box coordinates."""
+    def execute(self, agent_os: AgentOs) -> None:
+        agent_os.mouse(x=self.start_box.x, y=self.start_box.y)
+        time.sleep(0.2)
+        agent_os.click("left")
+
+class DoubleClickAction(BaseAction):
     action_type: Literal["left_double"] = "left_double"
     start_box: BoxCoordinate
 
-class RightClickAction(BaseModel):
-    """Right click action with start box coordinates."""
+    def execute(self, agent_os: AgentOs) -> None:
+        agent_os.mouse(x=self.start_box.x, y=self.start_box.y)
+        time.sleep(0.2)
+        agent_os.click('left')
+
+class RightClickAction(BaseAction):
     action_type: Literal["right_single"] = "right_single"
     start_box: BoxCoordinate
 
-class DragAction(BaseModel):
-    """Drag action with start and end box coordinates."""
+    def execute(self, agent_os: AgentOs) -> None:
+        agent_os.mouse(x=self.start_box.x, y=self.start_box.y)
+        time.sleep(0.2)
+        agent_os.click("right")
+
+class DragAction(BaseAction):
     action_type: Literal["drag"] = "drag"
     start_box: BoxCoordinate
     end_box: BoxCoordinate
 
-class HotkeyAction(BaseModel):
-    """Hotkey action with key combination."""
+    def execute(self, agent_os: AgentOs) -> None:
+        agent_os.mouse(x=self.start_box.x, y=self.start_box.y)
+        time.sleep(0.2)
```

Comment:
I would move this into the `AskUiControllerClient` as this from my perspective depends on the underlying `AgentOs` implementation and is the general sleep/wait time between actions to ensure the agent os implementation, os, application etc. had time to react. It is generally model-independent.

Also I would make this configurable through the constructor of `AskUiControllerClient` as it is highly dependent on os, application being automated etc.